### PR TITLE
Replace "Saved" indicator with an icon

### PR DESCRIPTION
**Pull request by susnux**
### 📝 Summary

* Part of #2836

Replaces the saving indicator with an icon, like the "active" users next to it. This also unifies the look on desktop and mobile devices.

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
desktop saving<br>desktop saved | new saving<br>new saved
mobile<br>mobile saving<br>mobile saved| same as above

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/master/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

**Comment by nimishavijay** [comment]
This is a great improvement! One suggestion: we can remove the background and make it look visually similar to the formatting options. 

I'm also thinking instead of the loading indicator we can use the floppy disk symbol, making it look the same as the save indicator on Office.

File: src/components/Editor/Status.vue [hunk 1]
```diff
@@ -22,13 +22,10 @@
...
-		<div v-if="$isMobile" v-tooltip="lastSavedStatusTooltip" :class="saveStatusClass" />
-		<div v-else
-			v-tooltip="lastSavedStatusTooltip"
-			class="save-status"
-			:aria-label="t('text', 'Document save status')"
-			:class="lastSavedStatusClass">
-			{{ lastSavedStatus }}
+		<div v-tooltip="lastSavedStatusTooltip" class="save-status" :class="saveStatusClass">
+			<CheckIcon v-if="saveStatusClass === 'saved'" />
+			<Loading v-else-if="saveStatusClass === 'saving'" />
```
[the remark this answers is not in the document]
Comment (susnux) [reply src/components/Editor/Status.vue:27]: Maybe we should then replace also the other usages of the loading icon within text with the icon from nextcloud-vue :)
But probably better on a follow up PR.

**Comment by susnux** [comment]
I pushed a commit to unify the look of the menu bar with the status & active user indicators:
* Set same size as menu bar entries: 44px
* Align icons vertically with menu bar
* Remove background and add background when hovering like menu bar

It looks now like this:

https://user-images.githubusercontent.com/1855448/207833335-e71efd0a-32f9-4541-954f-f3ea01c7a746.mp4

And BTW on error it would look like this:
saving indicator showing red warn sign

**Comment by raimund-schluessler** [comment]
Would be nice to align the saving indicator with the three-dots-button above.

**Review by max-nextcloud — APPROVED** [review]
Code looks good to me. From my point of view this is quite a big improvement.
I'd say this can be merged as is. Aligning the icons would be nice but i don't think it should block this. Using `NcLoadingIcon` everywhere was already postponed to a follow up PR.

**Comment by mejo-** [comment]
Looks great, thanks a lot @susnux. Given that sometimes it takes quite some time until the status is saved, I wonder whether the current rotating loading spinner is a bit too annoying for something that should be safe to ignore in most cases. What do you think=

**Comment by mejo-** [comment]
But I'm also fine with discussing this in a follow-up issue/PR.

**Comment by max-nextcloud** [comment]
> Looks great, thanks a lot @susnux. Given that sometimes it takes quite some time until the status is saved, I wonder whether the current rotating loading spinner is a bit too annoying for something that should be safe to ignore in most cases. What do you think=

Actually this is a good point. I think right now we only save every now and then to not overwhelm the server. So actually there are three states:
* saved
* modified but not saving yet
* saving right now

I think `modified` is currently treated as `saving` and shows the spinning wheel. Maybe we could show a pencil instead. Since that requires some additional logic i'd also opt for a follow up issue / pr.

**Comment by susnux** [comment]
Good point! And reading the comment by @jancborchardt mentioned above again, he already proposed using the dot indicator for unsaved changes!

Fixed this with the latest commit, it now looks like this:

https://user-images.githubusercontent.com/1855448/207884678-b095eaef-1af1-4ff7-9bc7-ebd53a1ec91b.mp4

**Comment by nimishavijay** [comment]
> I think modified is currently treated as saving and shows the spinning wheel. Maybe we could show a pencil instead. 

Suggestion: this icon used in Office
image
I can't find the MDI icon for this as it seems to be a custom icon, but it was suggested in the same comment by Jan that @susnux linked. We are trying to move towards MDI icons but this is used in Office and seems quite fitting as well :) 
cc @jancborchardt

**Comment by susnux** [comment]
> Would be nice to align the saving indicator with the three-dots-button above.

@raimund-schluessler While this is possible, it might be better to exchange the positions of the `current users` and the `saving indicator` and align the *saving* with the *close* and the *current users* with the *... menu*.
Because the *current user* button expands to the left if an user joins the session, so the alignment would be useless:

current user menu expanded to the left so alignment has no use

**Comment by susnux** [comment]
@nimishavijay Ok I think I misunderstood the comment, I thought we should use a check mark for saved and a dot for unsaved. But it should look like this (?):

https://user-images.githubusercontent.com/1855448/207905313-023e8cdd-ff76-4056-ac63-d0adb571844b.mp4

**Pull request merged**